packages/ag-charts-enterprise/src/box-plot/boxPlotSeries.ts

```diff
@@ -0,0 +1,383 @@
+                q3Value: Math.round(xAxis.scale.convert(q3Value, { strict: false })),
+                maxValue: Math.round(xAxis.scale.convert(maxValue, { strict: false })),
+            };
+
+            context.nodeData.push(nodeData);
+        });
+
+        return [context];
+    }
+
+    getLegendData(): _ModuleSupport.ChartLegendDatum[] {
+        return [];
+    }
+
+    // @ts-ignore
+    getTooltipHtml(seriesDatum: any): string {
+        return '';
+    }
+
+    protected isLabelEnabled(): boolean {
+        return false;
+    }
+
+    protected async updateDatumSelection(opts: {
+        nodeData: BoxPlotNodeDatum[];
+        datumSelection: _Scene.Selection<_Scene.Group, BoxPlotNodeDatum>;
+    }) {
+        const { nodeData, datumSelection } = opts;
+        const data = nodeData ?? [];
+        return datumSelection.update(data);
+    }
+
+    protected async updateDatumNodes(opts: {
+        datumSelection: _Scene.Selection<_Scene.Group, BoxPlotNodeDatum>;
+        isHighlight: boolean;
+    }) {
+        opts.datumSelection.each((group, datum) => {
+            const {
+                bandwidth,
+                minValue,
+                q1Value,
+                medianValue,
+                q3Value,
+                maxValue,
+                yValue,
+                fill,
+                fillOpacity,
+                stroke,
+                strokeWidth,
+                strokeOpacity,
+                lineDash,
+                lineDashOffset,
+            } = datum;
+
+            const selection = _Scene.Selection.select(group, _Scene.Rect);
+            const boxes = selection.selectByTag<_Scene.Rect>(GroupTags.Box);
+            const [outline] = selection.selectByTag<_Scene.Rect>(GroupTags.Outline);
+            const [median] = selection.selectByTag<_Scene.Line>(GroupTags.Median);
+            const whiskers = selection.selectByTag<_Scene.Line>(GroupTags.Whisker);
+            const caps = selection.selectByTag<_Scene.Line>(GroupTags.Cap);
+
+            outline.setStyles({ x: q1Value, y: yValue, width: q3Value - q1Value, height: bandwidth });
+
+            boxes[0].setStyles({
+                x: q1Value,
+                y: yValue,
+                width: Math.round(medianValue - q1Value + strokeWidth / 2),
+                height: bandwidth,
+            });
+
+            boxes[1].setStyles({
+                x: Math.round(medianValue - strokeWidth / 2),
+                y: yValue,
+                width: Math.floor(q3Value - medianValue + strokeWidth / 2),
+                height: bandwidth,
+            });
+
+            median.setStyles({
+                x: medianValue,
+                y1: yValue + strokeWidth,
+                y2: yValue + bandwidth - strokeWidth,
+            });
+
+            const capSize = 0.5;
+            const capY1 = yValue + bandwidth * capSize * 0.5;
+            const capY2 = yValue + bandwidth * capSize * 1.5;
```

Comment:
Nit: Could you extract the `0.5` and `1.5` numbers to constants to try and make them a little less magic here? Are these offsets in proportion of `capSize` (that's how I'm interpreting it)? I'm also presuming that the implied offset of `1` between the two is so that `capSize` is the actual width of the cap?

Reply:
I changed the implementation to not use constants